juntagrico/management/commands/mailtexts.py

```diff
@@ -14,13 +16,16 @@
 class Command(BaseCommand):
     # entry point used by manage.py
     def handle(self, *args, **options):
-        subscription = Subscription.objects.all()[0]
-        shares = Share.objects.all()[:2]
-        job = RecuringJob.objects.all()[0]
-        member = Member.objects.filter(MemberDao.has_future_subscription())[0]
-        member_wo_subs = Member.objects.filter(subscriptionmembership__isnull=True)[0]
-        co_member = Member.objects.filter(MemberDao.has_future_subscription())[1]
-        depot = Depot.objects.all()[0]
+        with transaction.atomic(durable=True):
+            # generate temporary test data to ensure that required objects are available
+            management.call_command('generate_testdata')
+            subscription = Subscription.objects.all()[0]
+            shares = Share.objects.all()[:2]
+            job = RecuringJob.objects.all()[0]
+            member, co_member = Member.objects.filter(MemberDao.has_future_subscription())[:2]
+            member_wo_subs = Member.objects.filter(subscriptionmembership__isnull=True)[0]
+            depot = Depot.objects.all()[0]
+            transaction.set_rollback(True)  # force rollback
```

Comment:
assume db crashes before the rollback is forced. the testdata might be found in a redo log of the db and be aplied
better way is to check if a correponding object exists in the db and if not use the new constructor without saving to create one

Reply:
foreign keys don't work without saving, at least not many to many and at least not the last time I tried that. So some of the templates fail.
Is this really how databases deal with uncompleted transactions after a crash? I would doubt it. Couldn't find any resource that supports yours nor my theory.
In any case I might try generating a temporary (in-memory) database to fill it with usable data for the email texts and use that. https://forum.djangoproject.com/t/using-an-in-memory-sqlite-database/472 or directly use the django test framework to achieve the same.